docs: MCP for Nondevs Blog (#1910)
block/goose: documentation/blog/2025-04-01-mcp-nondevs/index.md

---
title: "MCP Explained for Non-Developers"

> description: Learn what MCP or Model Context Protocol is and how anyone can use it to save time on tasks.

Reviewer: ```suggestion
description: Learn what Model Context Protocol (MCP) is and how anyone can use it to save time on tasks.
```

Now there are AI agents, or AI assistants, that can take actions and make decisions for you. But in order to have your AI agent interact with your systems, like Google Drive, Asana, or Slack, there wasn't a standard way to do it. At least not without figuring it out from scratch each time you needed your AI agent to work with what you need it to work with. That's super tedious.

That's exactly where MCP comes in. Best part is, you don't need to be a developer to start using them! MCP essentially allows you to give AI agents access to your external systems without having to code. You can think of MCP as the connector for a system and your AI agent, or like the USB-C of AI integrations.

## MCP Servers You Should Try Right Now
So what can you connect your AI agent to? MCP Servers! MCP servers give your agent access to your tools. With [over 3000 MCP servers](https://glama.ai/mcp/servers) you can connect to, here is your top list of popular MCP servers you should try:

> - **[Google Drive](https://block.github.io/goose/docs/tutorials/google-drive-mcp)**: File access and search capabilities for Google Drive

Reviewer: ```suggestion
- **[Google Drive](/docs/tutorials/google-drive-mcp)**: File access and search capabilities for Google Drive
```

> - **[YouTube Transcript](https://block.github.io/goose/docs/tutorials/youtube-transcript)**: Fetch and work with YouTube video transcripts

Reviewer: ```suggestion
- **[YouTube Transcript](/docs/tutorials/youtube-transcript)**: Grab and work with YouTube video transcripts
```

> - **[Google Maps](https://block.github.io/goose/docs/tutorials/google-maps-mcp)**: Location services, directions, and place details

Reviewer: ```suggestion
- **[Google Maps](/docs/tutorials/google-maps-mcp)**: Location services, directions, and place details
```

> - **[Tavily Web Search](https://block.github.io/goose/docs/tutorials/tavily-mcp)**: Web and local search using Tavily's Search API

Reviewer: ```suggestion
- **[Tavily Web Search](/docs/tutorials/tavily-mcp)**: Web and local search using Tavily's Search API
```

> - **[Asana](https://block.github.io/goose/docs/tutorials/asana-mcp)**: View asana tasks, projects, workspaces, and/or comments

Reviewer: ```suggestion
- **[Asana](/docs/tutorials/asana-mcp)**: View asana tasks, projects, workspaces, and/or comments
```

> - **[Speech](https://block.github.io/goose/docs/tutorials/speech-mcp)**: Real-time voice interaction, audio/video transcription, text-to-speech conversion and more

Reviewer: ```suggestion
- **[Speech](/docs/tutorials/speech-mcp)**: Real-time voice interaction, audio/video transcription, text-to-speech conversion and more
```

> - **[GitHub](https://block.github.io/goose/docs/tutorials/github-mcp)**: Tools to read, search, and manage Git repositories

Reviewer: ```suggestion
- **[GitHub](/docs/tutorials/github-mcp)**: Tools to read, search, and manage Git repositories
```

> - **[Fetch](https://block.github.io/goose/docs/tutorials/fetch-mcp)**: Web content fetching and conversion for efficient LLM usage

Reviewer: ```suggestion
- **[Fetch](/docs/tutorials/fetch-mcp)**: Web content fetching and conversion for efficient LLM usage
```

> This quick list should give you an idea of all the ways you can now use AI agents with your workflow. You can also explore community favorites in [handy MCP directories](https://dev.to/techgirl1908/my-favorite-mcp-directories-573n), and learn [how to check MCP servers are safe](https://block.github.io/goose/blog/2025/03/26/mcp-security) before installing.

Reviewer: ```suggestion
This quick list should give you an idea of all the ways you can now use AI agents with your workflow. You can also explore community favorites in [handy MCP directories](https://dev.to/techgirl1908/my-favorite-mcp-directories-573n), and learn [how to check MCP servers are safe](/blog/2025/03/26/mcp-security) before installing.
```

> You can also check out these [Goose tutorials](https://block.github.io/goose/docs/category/tutorials), showing you exactly how you can use some of these popular MCP servers with Goose, or use [Goose's Tutorial extension](https://block.github.io/goose/docs/tutorials/tutorial-extension) to get extra help walking you through using or building extensions.

Reviewer: ```suggestion
You can also check out these [Goose tutorials](/docs/category/tutorials), showing you exactly how you can use some of these popular MCP servers with Goose, or use [Goose's Tutorial extension](/docs/tutorials/tutorial-extension) to get extra help walking you through using or building extensions.
```

## The Possibilities Are Endless
While some are developed by official providers, a vast majority of MCP servers you see are actually developed by community members! Plus, because MCP is an open standard, anyone can build an MCP server for any resource. You could even use Goose to help you build one!

Hopefully now, instead of spending hours manually gathering data and creating your next marketing report, or manually sorting through your todo-backlog on a Monday, you will use MCP with Goose and have it done for you in minutes.

*To learn more about using MCP servers and Goose, check out the [Goose documentation](https://block.github.io/goose/docs/category/getting-started), or join the [Block Open Source Discord](https://discord.gg/block-opensource) to connect with other open source community members.*

<head>
  <meta property="og:title" content="MCP Explained for Non-Developers" />
  <meta property="og:type" content="article" />
  <meta property="og:url" content="https://block.github.io/goose/blog/2025/04/01/mcp-nondevs" />

>   <meta property="og:description" content="Learn what MCP or Model Context Protocol is and how anyone can use it to save time on tasks." />

Reviewer: ```suggestion
  <meta property="og:description" content="Learn what Model Context Protocol (MCP) is and how anyone can use it to save time on tasks." />
```

  <meta property="og:image" content="http://block.github.io/goose/assets/images/mcp_nondevs-5ce7f39de923cab01de6e14e5dc06744.png" />
  <meta name="twitter:card" content="summary_large_image" />
  <meta property="twitter:domain" content="block.github.io/goose" />
  <meta name="twitter:title" content="MCP Explained for Non-Developers" />

>   <meta name="twitter:description" content="Learn what MCP or Model Context Protocol is and how anyone can use it to save time on tasks." />

Reviewer: ```suggestion
  <meta name="twitter:description" content="Learn what Model Context Protocol (MCP) is and how anyone can use it to save time on tasks." />
```
